jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/KubernetesUndeployServiceTest.java

```diff
@@ -98,13 +98,16 @@ public void undeployWithManifestShouldDeleteApplicableEntities(@Mocked File file
     // Then
     // @formatter:off
     new Verifications() {{
-      kubernetesHelper.getKind((HasMetadata)any); times = 2;
+      kubernetesHelper.getKind((HasMetadata)any); times = 3;
```

Comment:
This test is used to verify that all *applicable* entities are deleted.

Previously this meant pod and service, but not namespace.

Please, add a new "not-applicable" entity to verify that it's **not** deleted.

Reply:
Hello, I am not able to understand	your comment. Could you please elaborate on it? Are you talking about a `jkubeServiceHub.getClient().resource(..)` verification entry for namespace? I think it's already there.

We were earlier filtering `Namespace` and `Project` in `undeployResources`. Now we're processing them after un-deploying standard resources. I'm not sure which entities can be considered as non-applicable since we're covering both now.

Reply:
No, I mean this test verifies that only "deletable"/"undeployable" instances are deleted.

Initially this test used Namespace, not because of its specific resource type, but because it was **not** deletable. `Namespace` resource verification is not what this test is for, it's to verify undeploy behavior.

This meant that the `kubernetesHelper.getKind((HasMetadata)any); ` method was called 2 times despite there was a request to delete 3 resources . This is because before this PR, Namespace was not an "undeployable" resource (now it is).
You need to find some other resource which is not "undeployable" and use that instead of Namespace.

Reply:
Umm, I'm sorry. I'm still a bit confused. Earlier we were ignoring `Namespace`/ `Project` which we're not doing anymore. Now we don't have any resource which is undeployable. Even if I include some resource that is not supported by resource API like this(`TokenReview` doesn't have any handler since it hasn't got any list type):
```java
  @Test
  public void undeployWithManifestShouldDeleteApplicableEntities(@Mocked File file) throws Exception {
    // Given
    final Namespace namespace = new NamespaceBuilder().withNewMetadata().withName("default").endMetadata().build();
    final TokenReview tokenReview = new TokenReviewBuilder().build(); // -> A Resource which can't be deployed via client.resource() API
    final Pod pod = new PodBuilder().withNewMetadata().withName("MrPoddington").endMetadata().build();
    final Service service = new Service();
    // @formatter:off
    new Expectations() {{
      file.exists(); result = true;
      file.isFile(); result = true;
      kubernetesHelper.loadResources(file);
      result = new HashSet<>(Arrays.asList(namespace, pod, service, tokenReview));
    }};
    // @formatter:on
    // When
    kubernetesUndeployService.undeploy(null, null, file);
    // Then
    // @formatter:off
    new Verifications() {{
      kubernetesHelper.getKind((HasMetadata)any); times = 4; // <- Extra invocation for TokenReview
      jKubeServiceHub.getClient().resource(pod).inNamespace("default")
              .withPropagationPolicy(DeletionPropagation.BACKGROUND).delete();
      times = 1;
      jKubeServiceHub.getClient().resource(service).inNamespace("default")
              .withPropagationPolicy(DeletionPropagation.BACKGROUND).delete();
      times = 1;
      jKubeServiceHub.getClient().resource(namespace)
              .withPropagationPolicy(DeletionPropagation.BACKGROUND).delete();
      times = 1;
    }};
    // @formatter:on
  }
```

`KubernetesHelper.getKind()` still gets invoked.